### ISSUE TITLE
Searching for a location in Autocomplete search bar and selecting it , makes the floating card shows loading indicator infinitely

### ISSUE THREAD

**Issue by maheshwara-bwd** [fysoul17/google_maps_place_picker#68]
The Floating card shows the loading indicator infinitely when i search for a location in search bar and selecting it.
I could see that the SearchingState is not set to Idle after a location is found.
But the floatingCard works perfectly when the pin is moved and the selected location is shown correctly

Can you pls provide a quick fix on this

Thanks in advance

**Comment by ZikunWang24** [comment]
I encountered this issue on iOS too. However, it worked fine on Android.

**Comment by maheshwara-bwd** [comment]
> I encountered this issue on iOS too. However, it worked fine on Android.

Yes, It works fine in android, only IOS have the issue

**Comment by ZikunWang24** [comment]
Any clue how to work around this?

**Comment by maheshwara-bwd** [comment]
> Any clue how to work around this?

Yes, i used a work around by using the 'selectedPlaceWidgetBuilder' and providing my own floating card.
In selectedPlaceWidgetBuilder function we can get the SearchingState and PickResult. The default floating card shows the Loading indicator when the state is Searching , In our case SearchingState is not changed to Idle, I did manually change the State to Idle if we have any data in PickResult. 
Hope this helps.

**Comment by ZikunWang24** [comment]
> > Any clue how to work around this?
> 
> Yes, i used a work around by using the 'selectedPlaceWidgetBuilder' and providing my own floating card.
> In selectedPlaceWidgetBuilder function we can get the SearchingState and PickResult. The default floating card shows the Loading indicator when the state is Searching , In our case SearchingState is not changed to Idle, I did manually change the State to Idle if we have any data in PickResult.
> Hope this helps.

Thanks so much!

**Comment by ZikunWang24** [comment]
I modified my code and it works fine most of the time, but sometimes it falsely thinks I am dragging the pin (even I am not) and stuck at the following situation:

![IMG_31C4E684DE2C-1](https://user-images.githubusercontent.com/35442448/93159531-36ffa280-f6c3-11ea-83f3-01f4e1ff8369.jpeg)


 I use something like this to check the state 

`selectedPlace == null
                                        ? Center(
                                            child: CircularProgressIndicator())
                                        : Column(
                                            ...`

Have you encountered something similar?

**Comment by maheshwara-bwd** [comment]
Yeah, i'm also facing the same issue, haven't found a solution yet.
Pls post here, if you do find any workarounds.

**Comment by alejandroaap** [comment]
I am facing the same problem. Any idea why it happens and how to fix it? Thank you

**Comment by alejandroaap** [comment]
The problem is that the call `provider.placeSearchingState = SearchingState.Idle` which is done after the` _moveTo (provider.selectedPlace.geometry.location.lat,
         provider.selectedPlace.geometry.location.lng); `is not notifying listeners to update status. The solution that I have done is to make a delay of 500 milliseconds and this has solved the problem.

Before:
`provider.placeSearchingState = SearchingState.Idle`
Then:
     `Future.delayed (Duration (milliseconds: 500), () => provider.placeSearchingState = SearchingState.Idle);`

@ fysoul17 Can you implement this solution or can you propose a different solution?

**Comment by alejandroaap** [comment]
The problem is that the call `provider.placeSearchingState = SearchingState.Idle` which is done after the` _moveTo (provider.selectedPlace.geometry.location.lat,
         provider.selectedPlace.geometry.location.lng); `is not notifying listeners to update status. The solution that I have done is to make a delay of 500 milliseconds and this has solved the problem.

Before:
`provider.placeSearchingState = SearchingState.Idle`
Then:
     `Future.delayed (Duration (milliseconds: 500), () => provider.placeSearchingState = SearchingState.Idle);`

@fysoul17 Can you implement this solution or can you propose a different solution?

**Comment by Abdullah-Zakhoi** [comment]
Hey @fysoul17, what do you think the problem is? ... I have the same issue.

> The `floating Card` shows the loading indicator infinitely when I search for a location in the search bar and selecting it.
I could see that the`SearchingState` is not set to Idle after a location is found.
But the`floating Card` works perfectly when the pin is moved and the selected location is shown correctly

**Comment by Abdullah-Zakhoi** [comment]
I fixed it by adding this 
``` 
          bool changeStateManually = false;
          if (selectedPlace != null){changeStateManually = true;}
          else{changeStateManually = false;}
```
in ` selectedPlaceWidgetBuilder` function.

and add `changeStateManually == false` with state condation will be
 `state == SearchingState.Searching  && changeStateManually == false`


full code of the function 
```
 selectedPlaceWidgetBuilder: (_, selectedPlace, state, isSearchBarFocused) {
          bool changeStateManually = false;
          if (selectedPlace != null){changeStateManually = true;}
          else{changeStateManually = false;}
          return isSearchBarFocused
              ? Container(color: Colors.red)
          // Use FloatingCard or just create your own Widget.
              : FloatingCard(
            bottomPosition: 0.0,    // MediaQuery.of(context) will cause rebuild. See MediaQuery document for the information.
            leftPosition: 0.0,
            rightPosition: 0.0,
            width: MediaQuery.of(context).size.width,
            height: 300,
            borderRadius: BorderRadius.circular(12.0),
            child: state == SearchingState.Searching  && changeStateManually == false?
            Center(child: Text('wait !!!')) :
            Column(
              children: [
                Text(selectedPlace.formattedAddress),
                RaisedButton(onPressed: () { print("do something with [selectedPlace] data"); },),
              ],
            ),
          );
        },
```

**Comment by kitoide** [comment]
I'm having a similar problem on Android, when I open the map and have my own selectedPlaceWidgetBuilder, when I click to search the load circle is triggered and the map reloads in my original position
